### DISCUSSION TITLE
Development status on LightTable

### DISCUSSION THREAD

**Discussion by prertik** [LightTable/LightTable#2506] [General]
Hi community 👋🏾 

You might have been wondering and seeing no updates on LightTable and such. And, I occasionally, do receive your emails or issues on what is happening or when will there be any new releases. I even created a repo to only discuss/raise issues on these queries but, it seems Discussions is a much better option.

So, most of you might have seen my talk on In/Clojure where I discussed a new design for LT and even gave a sneak peek of what I'm trying(though, there were major issues on that). I still am pursuing this, but, that will take more effort and time to cut it to a release. Again, there are most changes which I will just throw away after some time and so, I didn't turn the repo's visibility to the public yet. But, still, there are some ideas and things which we can apply to the current state of LightTable, yes, we have to call it a rewrite as, my target is to retain LightTable to its original glory with the addition of features which was discussed yet, didn't make it to LT's releases.  For this, I've discussed it numerous times, we have to embrace the open-sourced Clojure tools rather than fork and maintain a version of it. I've always been in awe of the vision and desire of [orchard](https://github.com/clojure-emacs/orchard). Due to orchard Clojure editor supports to editors like vscode is possible.

So, for the time being, when I say re-write I'm saying I want to take current LT but, try to embrace `orchard`, `nrepl` (upstream), yes, there can be a fork for some time but, it needs to be for short time and we should also give back to those communities, that way can only LT survive the test of time. Regarding my other vision of breaking LT into client-server architecture where we need not only support electron but, can be possible on multiple frontend is still a concept I believe in, but, I'll hold it off until we get a couple of LT releases based on our embrace of community.

So, when will this happen? What is the timeline?
This is a difficult question when we are in open-source, as, I'm a sole programmer working on LT and this is not my day job but, whatever free time I get, I try to work effectively on LT. As, like you, I'm also an admirer of LT. I also want to want to write my Clojure/Script code in LT rather than configure other tools to morph into that. I cannot give you the exact date, but, by my own time estimation based on the time I get, we might see a new release in summer 2021. Yes, it does read like a long time, but, we have waited years for a new release or activity, let's wait for some more.  What I can assure you is, after this release, we would be in a faster release cycle.

Thanks, Stay safe.

**Comment by ivarec** [comment]
What's bad about LightTable's current design? The code looks pretty neat :)

**Comment by prertik** [comment]
> So, when will this happen? What is the timeline?
> This is a difficult question when we are in open-source, as, I'm a sole programmer working on LT and this is not my day job but, whatever free time I get, I try to work effectively on LT. As, like you, I'm also an admirer of LT. I also want to want to write my Clojure/Script code in LT rather than configure other tools to morph into that. I cannot give you the exact date, but, by my own time estimation based on the time I get, we might see a new release in summer 2021. Yes, it does read like a long time, but, we have waited years for a new release or activity, let's wait for some more.  What I can assure you is, after this release, we would be in a faster release cycle.
> 
> Thanks, Stay safe.

Hi Community!

An update on how things are going, I have two works in progress of LT, one ofcourse is taking precedence over other. 1st one is using latest build tools and fixing parts which only needs fixing. This is a version, I intend to release very soon. I expect this to be out as RC this October(v 0.9.0).

2nd is, I'm overhauling the design and re-writing things keeping the foundation. So, this is a massive release maybe a version v1.0.0. I have been working and pushing changes in a separate (private) repo in LT as of now. I had created that for experimental things, but I have carried on  it, when I get it properly working, I'll pull changes in the main LT repo and release v0.9.0. I'm also hoping to accept contributions after I make push v0.9.0 and make foundational changes for v1.0.0. 

So, expect v0.9.0 to be a small release with not much new features, but lots of bug fixing release and v1.0.0 is going to be a massive one.

Thank you for sticking with this till now.

**Reply by aeonik** [reply]
Thanks for all your hard work @prertik! Looking forward to the new changes. I am new to Clojure, and your project helps inspire me to keep learning and building.
